lldb/include/lldb/ValueObject/DILAST.h

```diff
@@ -46,6 +46,13 @@ enum class CastKind {
   eNone,        ///< Type promotion casting
 };
 
+/// Promotions allowed for type casts in DIL.
+enum CastPromotionKind {
+  eArithmetic,
+  ePointer,
+  eNone,
```

Comment:
Can we comment on the individual enum cases what they represent. Maybe giving a small `C`/`C++` expression as an example